src/rust/engine/src/externs.rs

```diff
@@ -242,7 +238,7 @@ pub fn generator_send(generator: &Value, arg: &Value) -> Result<GeneratorRespons
         .into_iter()
         .zip(values.into_iter())
         .map(|(p, v)| Get {
-          product: *interns.insert_product(p).type_id(),
+          product: p,
```

Comment:
I might add a TODO here or on the python side to do more interning in `extern_generator_send()` in python so that we can centralize the process of interning in one place (I don't know how to do this yet given your comment on `Breaks` at https://github.com/pantsbuild/pants/pull/7114#discussion_r262816470, but it seems good to note maybe).